### PR TITLE
Upgrade PHPStan and use Composer v2

### PR DESCRIPTION
**Pull request by szepeviktor**
... and upgrade actions used in CI and phive packages.

### CODE REVIEW

File: src/dbless-wpdb.php [hunk 1]
```diff
@@ -135,7 +135,7 @@ public function close() {
 	}
 
 	public function has_cap( $db_cap ) {
-		return 1;
+		return true;
```
Comment (leogermani) [inline src/dbless-wpdb.php:138]: hmmm wonder why I had `1` there

**Review by leogermani — APPROVED** [review]
Thans for the house keeping!

File: src/dbless-wpdb.php (hunk 1 above)
Comment (szepeviktor) [reply src/dbless-wpdb.php:138]: I do not know that. Maybe we break the Internet with this change.
Core has booleans there: https://github.com/WordPress/WordPress/blob/588b0029558eae93b3dd359d4d14f226fb25329a/wp-includes/class-wpdb.php#L4031-L4083

Comment (szepeviktor) [reply src/dbless-wpdb.php:138]: You may take a look at coverage data and possible write a unit test for this method...

Comment (leogermani) [reply src/dbless-wpdb.php:138]: yeah, I checked core. Was wondering if there was any recent changes there.

Comment (szepeviktor) [reply src/dbless-wpdb.php:138]: In core 4.9 the docblock talked about an integer but **all** return values were boolean.
https://github.com/WordPress/WordPress/blob/4.9.22/wp-includes/wp-db.php#L3362

Comment (leogermani) [reply src/dbless-wpdb.php:138]: Maybe some ancient version of this method used `version_compare` without the third argument and expected a `1`? Anyway, we should be fine

**Comment by szepeviktor** [comment]
@leogermani Is there any hope this gets merged?

**Comment by leogermani** [comment]
Sorry @szepeviktor , I left it to you to do the honors since you're the author. Feel free to merge approved PRs (or don't you have permission?)

**Comment by szepeviktor** [comment]
> (or don't you have permission?)

I'm confused now 😵

**Pull request merged**